src/main/java/org/jabref/logic/integrity/AmpersandChecker.java

```diff
@@ -0,0 +1,40 @@
+package org.jabref.logic.integrity;
+
+import com.google.common.base.CharMatcher;
+import org.jabref.logic.l10n.Localization;
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.entry.field.Field;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.MatchResult;
+import java.util.regex.Pattern;
+
+/**
+ * Checks if the BibEntry contains unescaped ampersands.
+ */
+public class AmpersandChecker implements EntryChecker{
+    // matches for an & preceded by any number of \
+    private static final Pattern BACKSLASH_PRECEDED_AMPERSAND = Pattern.compile("\\\\*&");
+
+    @Override
+    public List<IntegrityMessage> check(BibEntry entry) {
+        List<IntegrityMessage> results = new ArrayList<>();
+
+        for (Map.Entry<Field, String> field : entry.getFieldMap().entrySet()) {
+            // counts the number of even \ occurrences preceding an &
+            long unescapedAmpersands = BACKSLASH_PRECEDED_AMPERSAND.matcher(field.getValue())
+                    .results()
+                    .map(MatchResult::group)
+                    .filter(m -> CharMatcher.is('\\').countIn(m) % 2 == 0)
+                    .count();
+
+            if (unescapedAmpersands > 0) {
+                results.add(new IntegrityMessage(Localization.lang("Found " + unescapedAmpersands + " unescaped '&'"), entry, field.getKey()));
```

Comment:
Ich glaube das geht so nicht. Die Localization-Klasse lädt statisch die Übersetztung aus einem Resource-Bundle. D.h. da dürfen keine dynamischen Infos mit drin sein. Das betrifft auch die Tests

Reply:
Man kann aber Parameter übergeben. Siehe:

```suggestion
                results.add(new IntegrityMessage(Localization.lang("Found %0 unescaped '&'", unescapedAmoersands), entry, field.getKey()));
```

Hab ich nicht getestet, aber im Code an anderen Stellen so gesehen

Reply:
Danke dir für die Info! Schaue ich mir gleich morgen genauer an.

Reply:
Die angefragten Änderungen wurden [angewendet](https://github.com/deybis17/jabref/pull/1/commits/5b110c2b3a6cd74ead7aec0604adec8db03b5b2c).